src/Arcus.Messaging.Health/Tcp/TcpHealthServer.cs

```diff
@@ -0,0 +1,161 @@
+using System;
+using System.Net;
+using System.Net.Sockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using GuardNet;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+
+namespace Arcus.Messaging.Health.Tcp
+{
+    /// <summary>
+    /// Representing a TCP server as a background process to expose an endpoint where the <see cref="HealthReport"/> is being broadcasted.
+    /// </summary>
+    public class TcpHealthServer : BackgroundService
```

Comment:
Suggestion, not required

```suggestion
    public class TcpHealthListener : BackgroundService
```

Reply:
Hmm, listener/speaker, client/server, sender/receiver, ... not sure.

Reply:
Server sounds too robust imo as it just listens for connections and not use TCP for the whole WebAPI, but that's just my opinion

Reply:
Renamed.